File: backend/Watcher.DataAccess/Entities/Instance.cs
namespace Watcher.DataAccess.Entities
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    
    public class Instance : Entity<int>
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public override int Id { get; set; }

        [Required]
        public string Address { get; set; }

        [Required]
        public string Platform { get; set; }

        public Guid GuidId { get; set; }

        public int OrganizationId { get; set; }
        public Organization Organization { get; set; }
        
        public IList<Dashboard> Dashboards { get; set; }
    }
}

File: frontend/src/app/shared/models/instance.ts
import { Organization } from './organization';
import { Dashboard } from './dashboard';

export class Instance {
  constructor(
    public id: number,
    public address: string,
    public platform: string,
    public organizationId: number,
    public organization: Organization,
    public dashboards: Dashboard[]) {
  }
}
